test-data/unit/check-protocols.test

```diff
@@ -2032,6 +2032,55 @@ reveal_type(list(b for b in B()))  # N: Revealed type is "builtins.list[__main__
 reveal_type(list(B()))  # N: Revealed type is "builtins.list[__main__.B]"
 [builtins fixtures/list.pyi]
 
+[case testOldStyleIterableOnClass]
+from typing import Tuple, TypeVar, Generic
+
+class CorrectGetItem:
+    def __getitem__(self, arg: int) -> str: pass
+class CorrectChild(CorrectGetItem): pass
+
+T = TypeVar('T')
+class WithGeneric(Generic[T]):
+    def __getitem__(self, arg: int) -> T: pass
+
+class ReturnsTuple:
+    def __getitem__(self, arg: int) -> Tuple[int, str]: pass
+
+class WrongGetItemSig:
+    def __getitem__(self, arg: str) -> str: pass
+
+reveal_type(list(a for a in CorrectGetItem()))  # N: Revealed type is "builtins.list[builtins.str]"
+reveal_type(list(a for a in CorrectChild()))  # N: Revealed type is "builtins.list[builtins.str]"
+
+x: WithGeneric[int]
+reveal_type(list(a for a in x))  # N: Revealed type is "builtins.list[builtins.int]"
+
+reveal_type(list(a for a in ReturnsTuple()))  # N: Revealed type is "builtins.list[Tuple[builtins.int, builtins.str]]"
+
+reveal_type(list(a for a in WrongGetItemSig()))  # N: Revealed type is "builtins.list[Any]" \
+                                                 # E: "WrongGetItemSig" has no attribute "__iter__" (not iterable)
```

Comment:
Hmm, this error message isn't ideal anymore, with this change. Could the error message be changed to just `"WrongGetItemSig" is not iterable`

Reply:
Error message are just insanely complex. I did my best in a new just-pushed iteration.
We can later further improve them.